pkg/subctl/cmd/gather.go

```diff
@@ -82,23 +85,43 @@ func gatherData() {
 	err := checkGatherArguments()
 	exitOnError("Invalid arguments", err)
 
-	for module, ok := range gatherModuleFlags {
-		if ok {
-			for dataType, ok := range gatherTypeFlags {
-				if ok {
-					status := cli.NewStatus()
-					status.Start(fmt.Sprintf("Gathering %s %s...", module, dataType))
-					status.End(cli.CheckForError(gatherFuncs[module](status, dataType)))
+	configs, err := getMultipleRestConfigs(kubeConfig, kubeContext)
+	exitOnError("Error getting REST config", err)
+	for _, item := range configs {
+		clientSet, err := kubernetes.NewForConfig(item.config)
+		exitOnError("error getting clientset", err)
+		clustername := item.clusterName
+		fmt.Println("Gathering information from", clustername)
+		for module, ok := range gatherModuleFlags {
+			if ok {
+				for dataType, ok := range gatherTypeFlags {
+					if ok {
+						status := cli.NewStatus()
+						status.Start(fmt.Sprintf("Gathering %s %s", module, dataType))
+						status.End(cli.CheckForError(gatherFuncs[module](status, clientSet, dataType, clustername)))
+					}
 				}
 			}
 		}
+		fmt.Println("")
 	}
 }
 
-func gatherConnectivity(status *cli.Status, dataType string) error {
+func gatherConnectivity(status *cli.Status, clientSet kubernetes.Interface, dataType string, clustername string) error {
```

Comment:
I suggest to use a struct called `gatherInfo` to contain the params. I suspect we may need to pass more params.

```suggestion
func gatherConnectivity(status *cli.Status, clientSet kubernetes.Interface, dataType string, clusterName string) error {
```

Reply:
Can we leave that to later improvement? I agree though.